plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -147,6 +151,21 @@ else if (!rangeConjuncts.isEmpty()) {
         return new ClusteringPushDownResult(domainsBuilder.build(), Joiner.on(" AND ").join(clusteringColumnPredicates));
     }
 
+    private static boolean rangesAreEquivalentToSingleValueNegation(List<Range> orderedRanges)
+    {
+        if (orderedRanges.size() != 2) {
+            return false;
+        }
+        Range first = orderedRanges.get(0);
+        Range second = orderedRanges.get(1);
+        return first.intersect(second).isEmpty()
+                && !first.isHighInclusive()
+                && !first.isHighUnbounded()
+                && !second.isLowInclusive()
+                && !second.isLowUnbounded()
+                && first.getHighBoundedValue().equals(second.getLowBoundedValue());
```

Comment:
Is it equivalent to `return domain.complement().isSingleValue()`?

Reply:
Yes that's what it should detect. I have a different solution however.